config/katex_replace/extension.rb

```diff
@@ -28,28 +28,37 @@ def process document, output
 
       katexScript = '<link rel="stylesheet" href="' + katexpath + '/katex.min.css">
 <script src="' + katexpath + '/katex.min.js"></script>
-<script src="' + katexpath + '/contrib/auto-render.min.js"></script>
-    <!-- Use KaTeX to render math once document is loaded, see
-         https://github.com/Khan/KaTeX/tree/master/contrib/auto-render -->
 <script>
     document.addEventListener("DOMContentLoaded", function () {
-        renderMathInElement(
-            document.body,
-            {
-                delimiters: [
-                    { left: "$$", right: "$$", display: true},
-                    { left: "\\\\\[", right: "\\\\\]", display: true},
-                    { left: "$", right: "$", display: false},
-                    { left: "\\\\\(", right: "\\\\\)", display: false}
-                ]
+        var elements, m;
+
+        elements = document.getElementsByClassName("latex-math-inline");
+        for (var i = 0; i < elements.length; ++i) {
+            m = elements[i];
+            katex.render(m.textContent, m, { displayMode: false });
+            if (m.childNodes.length == 1) {
+                m.parentNode.replaceChild(m.firstChild, m);
```

Comment:
Oops, I think it was actually just faster because it was causing the HTMLCollection (result of the getElementsByClassName) to change underneath us (as elements were removed). So it was only rendering half of the elements.